rir/src/compiler/translations/pir_2_rir.cpp

```diff
@@ -861,164 +794,91 @@ size_t Pir2Rir::compileCode(Context& ctx, Code* code) {
                 break;
             }
             case Tag::Subassign1_1D: {
-                // TODO: not tested
-                auto res = Subassign1_1D::Cast(instr);
-                load(it, res->arg<0>().val());
-                load(it, res->arg<1>().val());
-                load(it, res->arg<2>().val());
                 cs << BC::subassign1();
-                store(it, res);
                 break;
             }
             case Tag::Subassign2_1D: {
-                // TODO: not tested
                 auto res = Subassign2_1D::Cast(instr);
-                load(it, res->arg<0>().val());
-                load(it, res->arg<1>().val());
-                load(it, res->arg<2>().val());
                 cs << BC::subassign2(res->sym);
-                store(it, res);
                 break;
             }
             case Tag::Extract1_1D: {
-                // TODO: not tested
-                auto res = Extract1_1D::Cast(instr);
-                load(it, res->arg<0>().val());
-                load(it, res->arg<1>().val());
                 cs << BC::extract1_1();
-                store(it, res);
                 break;
             }
             case Tag::Extract2_1D: {
-                // TODO: not tested
-                auto res = Extract2_1D::Cast(instr);
-                load(it, res->arg<0>().val());
-                load(it, res->arg<1>().val());
                 cs << BC::extract2_1();
-                store(it, res);
                 break;
             }
             case Tag::Extract1_2D: {
-                // TODO: not tested
-                auto res = Extract1_2D::Cast(instr);
-                load(it, res->arg<0>().val());
-                load(it, res->arg<1>().val());
-                load(it, res->arg<2>().val());
                 cs << BC::extract1_2();
-                store(it, res);
                 break;
             }
             case Tag::Extract2_2D: {
-                // TODO: not tested
-                auto res = Extract2_2D::Cast(instr);
-                load(it, res->arg<0>().val());
-                load(it, res->arg<1>().val());
-                load(it, res->arg<2>().val());
                 cs << BC::extract2_2();
-                store(it, res);
                 break;
             }
             case Tag::IsObject: {
-                auto is = IsObject::Cast(instr);
-                load(it, is->arg<0>().val());
                 cs << BC::isObj();
-                store(it, is);
                 break;
             }
             case Tag::LdFunctionEnv: {
                 // TODO: what should happen? For now get the current env (should
                 // be the promise environment that the evaluator was called
                 // with) and store it into local and leave it set as current
-                if (!alloc.dead(instr)) {
-                    cs << BC::getEnv();
-                    store(it, instr);
-                }
+                cs << BC::getEnv();
```

Comment:
There is now `caller_env_`, can that be used here?

Reply:
no, that's a new env. the idea is that MkEnv creates an environment. So this just reloads the one, that was popped at the setEnv on 968. The idea is:

1. create env
2. set new created env as default env
3. set all variables
4. load env again, since this instruction should produce a value.

this is a new env, has nothing to do with parent_env